cmd/common/registry.go

```diff
@@ -216,6 +216,10 @@ func buildCommandFromGadget(
 			return cmd.ParseFlags(args)
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if showHelp, _ := cmd.Flags().GetBool("help"); showHelp {
```

Comment:
Ah, I just re-checked: these lines actually already exist [below](https://github.com/inspektor-gadget/inspektor-gadget/pull/1822/files#diff-9ae11145e0ddd34f2787a98a7b403de21c7315c218ef54b66440bc8d9cbd2818R240) - they just have to be moved [after (but outside) the `if parser != nil {` section](https://github.com/inspektor-gadget/inspektor-gadget/pull/1822/files#diff-9ae11145e0ddd34f2787a98a7b403de21c7315c218ef54b66440bc8d9cbd2818R244).

Sorry, can't do full suggestion in here with the change.

Reply:
[This](https://github.com/inspektor-gadget/inspektor-gadget/commit/3be19ad95552ae22805d6c7a8978663c51046b62) fixes it.

Reply:
ops right, it must be outside the conditional. Thanks!

Reply:
Sure, can do. Will fix the `--verbose` as well, then. I didn't want to hijack this, though.